api/scpca_portal/models/library.py

```diff
@@ -104,3 +109,33 @@ def get_formats_from_file_paths(cls, file_paths: List[Path]) -> List[str]:
         if any(path for path in file_paths if common.ANNDATA_EXT == path.suffix):
             formats.append(Library.FileFormats.ANN_DATA)
         return formats
+
+    @classmethod
+    def get_project_libraries_from_download_config(
+        cls, project, download_configuration: Dict
+    ) -> List[Self]:
+        if download_configuration not in common.VALID_PROJECT_DOWNLOAD_CONFIGURATIONS:
+            raise ValueError("Invalid download configuration passed. Unable to retrieve libraries.")
+
+        return [
+            library
+            for library in project.libraries.filter(modality=download_configuration["modality"])
+            if download_configuration["format"] in library.formats
+        ]
```

Comment:
I think we should add a test for this because `exclude_multiplexed` isn't accounted for yet.

Reply:
For merged something like this but you need to check the format requested and then check against the correct project attribute, `includes_merged_anndata` or `includes_merged_sce`. You might want to make an enum for this.
```python
if download_configuration["includes_merged"] and not project.includes_merged_anndata:
    return []
```